org.opent2t.sample.thermostat.superpopular/com.wink.thermostat/js/tests/test.js

```diff
@@ -2,37 +2,63 @@ var test = require('ava');
+console.log("Config:");
+console.log(JSON.stringify(config, null, 2));
+
 var translatorPath = require('path').join(__dirname, '..');
+var hubPath = require('path').join(__dirname, '../../../../org.opent2t.sample.hub.superpopular/com.wink.hub/js');
+
+var translator = undefined;
+
+function getThermostat(devices) {
+    for (var i = 0; i < devices.length; i++) {
+        var d = devices[i];
+
+        if (d.openT2T.translator === 'opent2t-translator-com-wink-thermostat') {
+            return d;
+        }
+    }
+
+    return undefined;
+}
+
+// setup the translator before all the tests run
+test.before(async t => {
+    var hubTranslator = await OpenT2T.createTranslatorAsync(hubPath, 'thingTranslator', config);
+    var hubInfo = await OpenT2T.getPropertyAsync(hubTranslator, 'org.opent2t.sample.hub.superpopular', 'HubResURI');
+    var deviceInfo = getThermostat(hubInfo.devices);
+
+    translator = await OpenT2T.createTranslatorAsync(translatorPath, 'thingTranslator', {'deviceInfo': deviceInfo, 'hub': hubTranslator});
+});
+
+test.serial("Valid Thermostat Translator", t => {
+    t.is(typeof translator, 'object') && t.truthy(translator);
+});
 
-///
-/// Run a series of tests to validate the translator
-///
+// 
+// Run a series of tests to validate the translator
+// 
 
 // Get AmbientTemperature via getter for individual property
 test.serial('AmbientTemperature', t => {
 
-    return OpenT2T.createTranslatorAsync(translatorPath, 'thingTranslator', config.Device)
-        .then(translator => {
-            // TEST: translator is valid
-            t.is(typeof translator, 'object') && t.truthy(translator);
-            return OpenT2T.getPropertyAsync(translator, 'org.opent2t.sample.thermostat.superpopular', 'ambientTemperature')
-                .then((ambientTemperature) => {
+    return OpenT2T.getPropertyAsync(translator, 'org.opent2t.sample.thermostat.superpopular', 'ambientTemperature')
+        .then((ambientTemperature) => {
 
-                    // TEST: some ambient temperature was returned
-                    console.log('*** ambientTemperature: ' + ambientTemperature);
-                    t.truthy(ambientTemperature);
-                });
+            // TEST: some ambient temperature was returned
+            console.log('*** ambientTemperature: ' + ambientTemperature);
+            t.truthy(ambientTemperature);
         });
 });
 
 // Set/Get TargetTemperatureHigh via setters for individual properties
 test.serial('TargetTemperatureHigh', t => {
 
-    return OpenT2T.createTranslatorAsync(translatorPath, 'thingTranslator', config.Device)
-        .then(translator => {
-            // TEST: translator is valid
-            t.is(typeof translator, 'object') && t.truthy(translator);
+    return OpenT2T.getPropertyAsync(translator, 'org.opent2t.sample.thermostat.superpopular', 'targetTemperatureHigh')
+        .then((targetTemperatureHigh2) => {

```

Comment:
>targetTemperatureHigh2 [](start = 15, length = 22)

It's a little weird that targetTemperatureHigh2 comes before targetTemperatureHigh.

Consider using a more descriptive variable name.